lib/eventstream_rpc.ts

```diff
@@ -837,6 +844,11 @@ export class RequestResponseOperation<RequestType, ResponseType> extends EventEm
                 await this.operation.activate(requestMessage);
 
                 let message : eventstream.Message = await responsePromise;
+
+                // If the server terminated the stream, then set the operation to be ended immediately
+                if ((message.flags ?? 0) & eventstream.MessageFlags.TerminateStream) {
```

Comment:
Do we need to deserializeResponse (line 852) prior to checking `message.flags` ?

Reply:
nvm. It looks like we're good. Was thrown off by the different ordering here vs. what's going on below on 941

Reply:
No, the message flags are already available before deserialization. Deserialization is only for the application-level data. I can switch the order if you think that's better though.